### DISCUSSION TITLE
作业这个特性可以往生态的方向建设

### DISCUSSION THREAD

**Discussion by HansBug** [MaaAssistantArknights/MaaAssistantArknights#1353] [Ideas]
比较直接的一点，自动作业这功能确实很不错。不过眼下直接打开一个飞机驾驶室一样的页面再来回操作，emm，感觉还是可以继续改进的。

我个人的一些脑洞：
1. 现在的作业编写类似于鼠标拖块编程的感觉，而实际上可以考虑提供类似“宏录制”的模式，**直接让作业编写者在真实游戏/游戏模拟器上进行操作**，系统负责录下完整过程并生成JSON数据。
2. 当然，更具体的话，考虑到mrfz作业和单纯的宏录制还是略有不同的，因此可以考虑加上一些**针对游戏的处理逻辑**。比如真银斩sp满了直接开技能，费用够了立刻下42、闪避怪清掉后撤干员等，这些东西对不同box的玩家可能略有差异，甚至可能看脸（比如闪避、闪避还是闪避 🤣 ），而且经常导致翻车。
3. 回到1，游戏地图其实现在已经有开源平台做了（https://map.ark-nights.com/map/ ），可以直接获取地图信息。
4. 还是1，之前看到过另一个工具站（名字忘了），上面可以搜关卡作业，这个没准可以联动一下，甚至直接使之提供JSON下载。
5. 可以建设一个作业平台，让更多人愿意产出JSON，也让更多的人能直接获取到现成的JSON。
6. 基于5，脑洞更大一些的话，可以考虑做点统计，比如**根据一份作业在实机上的表现，量化统计一波作业的速刷能力、稳定性**等数据信息，这对抄作业的来说非常关键，而目前却没见到过有效的应对手段。
7. 更进一步，如果能搜集更多的信息的话，考虑DM走一波，没准有收获，甚至可能出paper。

总之我个人感觉，这个东西可以往生态的方面去建设，让更多的人来用这个作业功能。不说别的，如果可以做到直接在软件里头选择关卡，完后直接就开打，这就真的厉害了。

一点个人的胡思乱想~~~ 😸

**Comment by MistEO** [comment]
感谢大佬，想法非常好，然后

1. 这个我们之前有认真考虑过，实际要写需要考虑的东西非常多，难度有点大，可能短期内弄不了
2. 同 1 ?
3. 我们打算自己做这样一个平台，直接在网页上点击地图就好了，但是现在很缺前端 ~~工具人~~ 大佬帮忙干活。短期内的计划是我们在本地导出所有地图并把坐标标好，前端上直接获取已经生成好的图片显示出来
4. 同上
5. 同 3, 已经有计划了，甚至 [repo](https://github.com/MaaAssistantArknights/MaaCopilotServer) 都开好了，但是缺人手做
6. 有这个计划，还是缺人
7. 缺人

QAQ 所以如果有大佬想加入我们，那 gkd！！！

**Comment by HansBug** [comment]
嗯哼，感谢回复。

1、2的话，考虑到做个全真的模拟器确实工程量不小，除非有办法反编译出官方代码实现（且官方代码设计良好，不能耦合过大），不然能利用的只有**关卡解包数据**。或者就在官方游戏上运行，adb尝试**录制一下视频，再根据视频识别操作**？（cv方向我也没那么熟，just brainstorming 🤣 ，只是感觉实际作业视频中各种行为特征其实还挺明显的；**这个路线要是实现了的话，那么b站上现有视频都可以考虑转化为JSON作业**，直接起飞）
其实我理解缺人的情况，我本人也感兴趣不过对前端是真不熟（本人软工专业，主要搞工具链、基础设施，主要语言Python/Java/Ruby，前端技术栈停留在jquery时代 😢 ）。因此就短期内，其实你们可以考虑**把一些热点关卡**（比如SV-7/8/9、SV-EX-8、比较新的悖论模拟等）的**JSON作业做好后整理到repo wiki上**，也可以发动小圈子内的人搞一波，再给下个小版本里面加个面向repo wiki的自动获取功能，先把这个功能给支棱起来，实现一个MVP（Minimum Viable Product）。
生态建设这部分，没准可以参考企鹅物流那边的玩法，我猜他们也有同样的问题：这是个数据驱动的站，一开始没几个人用，没数据，因此信息不准确，于是更没人愿意用。不过后来他们还是成功破局了，而且很大一部分数据来源于自动上传。

以上~~

**Comment by zzyyyl** [comment]
这是不是基本上 complete 了？
